packages/rocketchat-spotify/spotify.coffee

```diff
@@ -0,0 +1,48 @@
+###
+# Spotify a named function that will process Spotify (ex: spotify:track:1q6IK1l4qpYykOaWaLJkWG)
+# @param {Object} message - The message object
+###
+
+class Spotify
+	process = (message, source, callback) ->
+		if _.trim source
+			# Separate text in code blocks and non code blocks
+			msgParts = source.split(/(```\w*[\n\ ]?[\s\S]*?```+?)/)
+
+			for part, index in msgParts
+				# Verify if this part is code
+				codeMatch = part.match(/```(\w*)[\n\ ]?([\s\S]*?)```+?/)
+				if not codeMatch?
+					callback message, msgParts, part, index
```

Comment:
This could result in the callback being called multiple times. Is this intended?